src/providers/ldap/sdap_async_nested_groups.c

```diff
@@ -2298,18 +2341,22 @@ sdap_nested_group_deref_direct_process(struct tevent_req *subreq)
     struct sdap_options *opts = NULL;
     struct sdap_deref_attrs **entries = NULL;
     struct ldb_message_element *members = NULL;
+    hash_table_t *members_set = NULL; /* will be used as a `set` */
     const char *orig_dn = NULL;
-    const char *member_dn = NULL;
     size_t num_entries = 0;
-    size_t i, j;
-    bool member_found;
+    size_t i;
     errno_t ret;
 
     req = tevent_req_callback_data(subreq, struct tevent_req);
     state = tevent_req_data(req, struct sdap_nested_group_deref_state);
 
     opts = state->group_ctx->opts;
     members = state->members;
+    members_set = convert_ldb_element_to_set(state->members);
+    if (!members_set) {
```

Comment:
Hi,

I would prefer `members_set == NULL` here (see e.g. https://blog.cryptomilk.org/2013/03/28/writing-and-reading-code/).

bye,
Sumit

Reply:
Thanks for the link.
Funny part is that second code snippet example in this article has `if (!ok) {`
Frankly, I don't buy the reasoning provided, but no problems - I'll change it.

Reply:
Done.